docs/src/pages/docs/integrations/css.md

```diff
@@ -82,6 +82,8 @@ export default {
       p3: true,
       /** normalize all colors */
       colorFormat: 'hex',
+      /** used to replace whitespace in the middle of group and token names */
+      spaceReplacement: '_',
```

Comment:
Thanks so much for this PR! But what are your thoughts on an alternative API that may be more flexible? It would let people specify different styles, e.g.

```suggestion
      variableName: (name) => camelcase(name)
```

I think we should still by default handle spaces in tokens. But in case people wanted to override that, they could pass in a function to enforce a specific style, e.g. `camelCase` or `snake_case` or `kebab-case` (all valid in CSS).

Reply:
Semi-related thought—I’m not super opinionated either on the default behavior, but I have noticed the W3C Design Token format prefers `camelCase` style when possible. Should we perhaps convert tokens with spaces to `camelCase` (and **only spaces**, to be clear)? e.g.

- `my-token` → `my-token` (same)
- `my token` → `myToken` (camelcased)

I’m just guessing at what would match the theoretical W3C styleguide the closest (which isn’t explicated, but just thinking out load)

Reply:
> what are your thoughts on an alternative API that may be more flexible? It would let people specify different styles

I could see that being useful to some people. Shouldn't be difficult to implement either. It does open up the possibility that the user can actually break middle space handling, but as long as we document it and handle it by default I think that's ok. Two questions on the proposed alternative.

1. You named the option `variableName` in your example.  This API will have to handle both variable and group names. Do you think that'll be understood by the user if we document it or should we come up with a different name?
2. Just double checking, with this approach we'd be running this option on every variable/group name right? Not just when we need to deal with a middle space?

> I’m not super opinionated either on the default behavior, but I have noticed the W3C Design Token format prefers camelCase style when possible.

> I’m just guessing at what would match the theoretical W3C styleguide the closest

Where have you noticed the `camelCase` style?  Looking through [the design tokens spec](https://tr.designtokens.org/format/) the few examples that match this issue all seem to use `kebab-case`.

Reply:
> Where have you noticed the `camelCase` style?

Mostly a historical change, from the standard moving from `kebab-case` to `camelCase` with things like `$type` changes ([PR](https://github.com/design-tokens/community-group/pull/111)). You’re right that most of the examples still haven’t changed from the original `kebab-case`, but the spec itself has decidedly chosen `camelCase`

Reply:
And to answer your 2 questions:

> 1. You named the option variableName in your example. This API will have to handle both variable and group names. Do you think that'll be understood by the user if we document it or should we come up with a different name?

I don’t feel too strongly on the name here. I was originally thinking it’d be easier to first generate a **valid CSS identifier** for each token output, but let the user override it, e.g.

```ts
pluginCSS({
  variableName((name) => {
    // var(--token-i-dont-like);
    if (name === 'token-i-dont-like') {
      return 'better-name';
    }
    return name;
  }),
})
```

Also, if we wanted to provide additional context, we could do something like:

```ts
// "token" is the original token
variableName((name, token) => {
```

But I’m open to other suggestions if you had a different API in mind! But overall I was thinking that with the addition of the `spaceReplacement` option, we’re providing a 2nd option to control naming **without letting the user actually control it** (the other being `prefix`). I’m anticipating other scenarios besides whitespace that a user may want to override a CSS variable name (e.g. perhaps they have some weird library interaction, or conflict, they need to handle) so turning this into a callback function would solve #123 while solving other future problems in one change.

> 2. Just double checking, with this approach we'd be running this option on every variable/group name right? Not just when we need to deal with a middle space?

That’s my thought, but again, open to pushback. I was thinking we’d pass back a **finished, valid CSS identifier** so we’d solve the #123 bug for users automatically. But let the user customize that if they wanted, for whatever reason.

Reply:
Love it! That’s exactly solving the problems I was thinking about, and then some. I agree with everything you proposed, and would like to implement it as-suggested. I only have a few minor thoughts which don’t change things too much:

- I would like to provide `token` as a 2nd parameter to `transformName`—that way users can choose to make different decisions, say, based on `$type` (most can ignore it, but I can think of a few scenarios where you may want context of the original token)
- Regarding “returning the full css variable name (`--` included)”, I’m fine with that but would probably also allow them to omit that if they’d like (i.e. add it if missing). Reason is I can’t imagine a scenario where you’d truly intend for `--` to be omitted? But happy to revisit if that’s not the case
- Regarding `prefix`: let’s keep for now so as we don’t have to release a breaking version of the plugin. But for v2.0 I’d drop `prefix` for sure and keep only `transformName`.